src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -0,0 +1,236 @@
+    "faldo" -> "http://biohackathon.org/resource/faldo#",
+    "metatype" -> "https://w3id.org/biolink/biolinkml/type/",
+    "owl" -> "http://www.w3.org/2002/07/owl#",
+    "pav" -> "http://purl.org/pav/",
+    "qud" -> "http://qudt.org/1.1/schema/qudt#",
+    "rdf" -> "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+    "rdfs" -> "http://www.w3.org/2000/01/rdf-schema#",
+    "shex" -> "http://www.w3.org/ns/shex#",
+    "skos" -> "https://www.w3.org/TR/skos-reference/#",
+    "void" -> "http://rdfs.org/ns/void#",
+    "wgs" -> "http://www.w3.org/2003/01/geo/wgs84_pos",
+    "xsd" -> "http://www.w3.org/2001/XMLSchema#",
+    "rdf" -> "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+    "rdfs" -> "http://www.w3.org/2000/01/rdf-schema#",
+    "go" -> "http://www.geneontology.org/formats/oboInOwl#",
+    "blml" -> "https://w3id.org/biolink/biolinkml/meta/",
+    "bl" -> "https://w3id.org/biolink/vocab/",
+    "MONDO" -> "http://purl.obolibrary.org/obo/MONDO_",
+    "EMAPA" -> "http://purl.obolibrary.org/obo/EMAPA_",
+    "SO" -> "http://purl.obolibrary.org/obo/SO_",
+    "RO" -> "http://purl.obolibrary.org/obo/RO_",
+    "GO" -> "http://purl.obolibrary.org/obo/GO_",
+    "CHEBI" -> "http://purl.obolibrary.org/obo/CHEBI_",
+    "BFO" -> "http://purl.obolibrary.org/obo/BFO_",
+    "obo" -> "http://purl.obolibrary.org/obo/",
+    "NCBIGENE" -> "http://identifiers.org/ncbigene:",
+    "sesame" -> "http://www.openrdf.org/schema/sesame#",
+    "owl" -> "http://www.w3.org/2002/07/owl#",
+    "prov" -> "http://www.w3.org/ns/prov#",
+    "MESH" -> "http://id.nlm.nih.gov/mesh/"
+  )
+
+  def makeHttpClient: UIO[TaskManaged[Client[Task]]] =
+    ZIO.runtime[Any].map { implicit rts =>
+      BlazeClientBuilder[Task](rts.platform.executor.asEC).withConnectTimeout(Duration(3, MINUTES)).resource.toManaged
+    }
+
+  implicit val sparqlJsonDecoder: EntityDecoder[Task, ResultSet] = EntityDecoder[Task, String].flatMapR { jsonText =>
+    Try(ResultSetFactory.fromJSON(IOUtils.toInputStream(jsonText, StandardCharsets.UTF_8))) match {
+      //IntelliJ shows false compile errors here; check with SBT
+      //Requires import of ZIO-Cats interop typeclasses to compile
+      case Success(resultSet) => DecodeResult.success(resultSet)
+      case Failure(e) => DecodeResult.failure(MalformedMessageBodyFailure("Invalid JSON for SPARQL results", Some(e)))
+    }
+  }
+
+  def getNodeTypes(nodes: List[KGSNode]): Map[String, String] = {
+    val nodeTypes = nodes.collect {
+      case (node) if node.`type`.nonEmpty => (node.id, "bl:" + CaseUtils.toCamelCase(node.`type`, true, '_'))
+    }.toMap
+    val newNodeTypes = nodeTypes ++ nodes.flatMap(node => node.curie.map(node.id -> _)).toMap
+    newNodeTypes
+  }
+
+  def runSPARQLSelectQuery(query: String, appConfig: AppConfig): Task[ResultSet] =
+    for {
+      clientManaged <- makeHttpClient
+      _ = logger.debug("query: {}", query)
+      //building a uri from config shouldn't be this verbose...any better way to do this?
```

Comment:
I agree, this seems really verbose. I haven't worked with `Uri` enough to say if there is a better approach. But also I don't think we need to specify these pieces separately. It would be fine for the config to just accept a full URL pointing to the SPARQL endpoint. That would even have some advantages since here we need to make assumptions about the URL path (e.g. `blazegraph/sparql`)  to add to the host to where the SPARQL service is. There isn't anything constraining that, and it could be completely different.